swift/ql/lib/codeql/swift/elements/expr/MethodLookupExpr.qll

```diff
@@ -14,7 +14,16 @@ class MethodLookupExpr extends Generated::MethodLookupExpr {
 
   override Decl getImmediateMember() {
     result =
-      Synth::convertDeclFromRaw(this.getUnderlying().getFunction().(Raw::DeclRefExpr).getDecl()) // TODO: FIX THIS
+      Synth::convertDeclFromRaw([
+          this.getUnderlying().getFunction().(Raw::DeclRefExpr).getDecl(),
+          this.getUnderlying()
+              .getFunction()
+              .(Raw::FunctionConversionExpr)
+              .getSubExpr()
+              .(Raw::DeclRefExpr)
+              .getDecl(),
+          this.getUnderlying().getFunction().(Raw::OtherConstructorDeclRefExpr).getConstructorDecl()
+        ])
```

Comment:
this seems a bit ad-hoc, we should probably account for any number of implicit conversions just in case maybe? Something like

```codeql
private Raw::Expr getConverted(Raw::Expr e) { result = e.(Raw::ImplicitConversionExpr).getSubExpr() }

...
  override Decl getImmediateMember() {
    exists(Raw::Expr f |
      f = getConverted*(this.getUnderlying().getFunction()) and
      result = Synth::convertDeclFromRaw([
          f.(Raw::DeclRefExpr).getDecl(),
          f.(Raw::OtherConstructorDeclRefExpr).getConstructorDecl()
      ])
   )
```

Reply:
@redsun82 it seems like we're reinventing the "hidden" AST with your solution. Can we avoid that somehow?

Reply:
Well, `MethodLookupExpr` is a purely synthesized class, so we may be "overriding" these predicates, but their `super.` is in fact empty, since there is no `@method_lookup_expr` in the database. So we're just populating them for the first time based on the underlying `Raw::SelfApplyExpr`.

I agree that, just to be safe, I should add some transitive closures here.

Reply:
> @redsun82 it seems like we're reinventing the "hidden" AST with your solution. Can we avoid that somehow?

@MathiasVP:
Yeah, I was also turning my nose up while writing that, thinking the same. The problem is that that hidden AST is implemented on the `Generated` layer, while here as @d10c correctly points out we are actually populating this and to do so we must move in the `Raw` layer. I don't think moving the hidden ast to `Raw` would really be advisable, so I was a bit at a loss for a better solution.

Maybe a possibility would be to move the specification of conversions in the schema as a property option, and then make that available to more generated layers...

Reply:
> I don't think moving the hidden ast to Raw would really be advisable,

Why not?

Reply:
because we would need to also keep it in the public layer as well, in case some sinthesized nodes need handling for hiding. We could have it in both layers, but that might be a bit confusing... but maybe there's a way.

Reply:
I can imagine that this probably one of many synthesized AST nodes we'd want to have. So having a proper way to handle these conversions when constructing synthedized AST nodes would be very helpful. No need to do in for this PR, of course. I'll create a ticket for it.